template/template_test.go

```diff
@@ -1606,8 +1607,87 @@ func TestTemplate_Execute(t *testing.T) {
 			"1",
 			false,
 		},
+		{
+			"leaf_cert",
+			&NewTemplateInput{
+				Contents: `{{with caLeaf "foo"}}` +
+					`{{.CertPEM}}{{.PrivateKeyPEM}}{{end}}`,
+			},
+			&ExecuteInput{
+				Brain: func() *Brain {
+					d := dep.NewConnectLeafQuery("foo")
+					b := NewBrain()
+					b.Remember(d, &api.LeafCert{
+						Service:       "foo",
+						CertPEM:       "PEM",
+						PrivateKeyPEM: "KEY",
+					})
+					return b
+				}(),
+			},
+			"PEMKEY",
+			false,
+		},
+		{
+			"root_ca",
+			&NewTemplateInput{
+				Contents: `{{range caRoots}}{{.RootCertPEM}}{{end}}`,
+			},
+			&ExecuteInput{
+				Brain: func() *Brain {
+					d := dep.NewConnectCAQuery()
+					b := NewBrain()
+					b.Remember(d, []*api.CARoot{
+						&api.CARoot{
+							Name:        "Consul CA Root Cert",
+							RootCertPEM: "PEM",
+							Active:      true,
+						},
+					})
+					return b
+				}(),
+			},
+			"PEM",
+			false,
+		},
+		{
+			"func_connect",
+			&NewTemplateInput{
+				Contents: `{{ range connect "webapp" }}{{ .Address }}{{ end }}`,
+			},
+			&ExecuteInput{
+				Brain: func() *Brain {
+					b := NewBrain()
+					d, err := dep.NewHealthServiceQuery("webapp")
```

Comment:
```suggestion
					d, err := dep.NewHealthConnectQuery("webapp")
```
I suspect your setup will need to be adjusted after you actually use the new code.

Reply:
:man_facepalming:

Reply:
And actually no other changes are needed, aside from calling the correct method. Those tests are totally mock based using the data injected into the *Brain (thinking Neo plugging in here).. and that data was correct. So as long as the NewHealthConnectQuery itself doesn't return an error it behaves the same.

Reply:
Fixed